### ISSUE TITLE
CronJob API version update to batch/v1

### ISSUE THREAD

**Issue by britdm** [uselagoon/build-deploy-tool#129]
https://github.com/uselagoon/build-deploy-tool/blob/a0a2b34c3e9fe08982709e8c4f8b0e07b05de684/legacy/helmcharts/cli/templates/cronjob.yaml#L3

The apiVersion for CronJobs in `legacy/helmcharts/cli/templates/cronjob.yaml` references the depreciated `batch/v1beta1` version, which is no longer supported in v1.22+. Requesting an update to switch to using `batch/v1` that is supported in active and maintained Kubernetes versions.

**Comment by britdm** [comment]
Adding reference to the `cronjob.yaml` in `basic/templates/cronjob.yaml` in https://github.com/uselagoon/build-deploy-tool/blob/main/legacy/helmcharts/basic/templates/cronjob.yaml

**Comment by shreddedbacon** [comment]
Will discuss with the team, but v1beta1 is available until kubernetes 1.25 https://kubernetes.io/docs/reference/using-api/deprecation-guide/#cronjob-v125

**Comment by tobybellwood** [comment]
Lagoon v2.12.x (released in Jan 2023) will support a minimum Kubernetes version of 1.21, so this change will be introduced then.  For anyone needing backwards compatibility after that point :scream:, they can always fork and add a semver comparison in there
```
{{- if semverCompare ">=1.21-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: batch/v1
{{- else -}}
apiVersion: batch/v1beta1
{{- end }}
```
